### ISSUE TITLE
JHipster v5.7.1 - Apllication run fails due to the org.springframework.beans.factory.BeanDefinitionStoreException

### ISSUE THREAD

**Issue by msshitole** [jhipster/generator-jhipster#8964]
**JHipster INFO**
INFO! Using JHipster version installed locally in current project's node_modules
INFO! Executing jhipster:info
INFO! Options: from-cli: true
Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**

```
myapp@0.0.0 /home/shitole/myapp
└── generator-jhipster@5.7.1 

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
    "generator-jhipster": {
        "promptValues": {
            "packageName": "com.mycompany.myapp",
            "nativeLanguage": "en"
        },
        "jhipsterVersion": "5.7.1",
        "applicationType": "monolith",
        "baseName": "myapp",
        "packageName": "com.mycompany.myapp",
        "packageFolder": "com/mycompany/myapp",
        "serverPort": "8080",
        "authenticationType": "jwt",
        "cacheProvider": "ehcache",
        "enableHibernateCache": true,
        "websocket": false,
        "databaseType": "sql",
        "devDatabaseType": "h2Memory",
        "prodDatabaseType": "mysql",
        "searchEngine": false,
        "messageBroker": false,
        "serviceDiscoveryType": false,
        "buildTool": "maven",
        "enableSwaggerCodegen": false,
        "jwtSecretKey": "bXktc2VjcmV0LXRva2VuLXRvLWNoYW5nZS1pbi1wcm9kdWN0aW9uLWFuZC10by1rZWVwLWluLWEtc2VjdXJlLXBsYWNl",
        "clientFramework": "angularX",
        "useSass": true,
        "clientPackageManager": "npm",
        "testFrameworks": ["cucumber"],
        "jhiPrefix": "jhi",
        "otherModules": [],
        "enableTranslation": true,
        "nativeLanguage": "en",
        "languages": ["en"]
    }
}

</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>

##### **Environment and Tools**

openjdk version "1.8.0_191"
OpenJDK Runtime Environment (build 1.8.0_191-8u191-b12-0ubuntu0.18.04.1-b12)
OpenJDK 64-Bit Server VM (build 25.191-b12, mixed mode)

git version 2.19.1

node: v8.11.0

npm: 6.4.1

yeoman: 2.0.5

yarn: 1.10.1

Docker version 18.06.1-ce, build e68fc7a

docker-compose version 1.23.1, build b02f1306

INFO! Congratulations, JHipster execution is complete!

 **Overview of the issue**

I have just generated simple myapp monolith application. And when I run the applications with command "./mvnw" to see the app and explore APIs on http://localhost:8080.

Here is the complete stackTrace for more details

```
2018-12-17 13:38:40.779 ERROR 8038 --- [  restartedMain] o.s.boot.SpringApplication               : Application run failed

org.springframework.beans.factory.BeanDefinitionStoreException: Failed to process import candidates for configuration class [springfox.documentation.swagger2.configuration.Swagger2DocumentationConfiguration]; nested exception is java.io.FileNotFoundException: class path resource [com/google/common/base/Supplier.class] cannot be opened because it does not exist
	at org.springframework.context.annotation.ConfigurationClassParser.processImports(ConfigurationClassParser.java:646)
	at org.springframework.context.annotation.ConfigurationClassParser.doProcessConfigurationClass(ConfigurationClassParser.java:303)
	at org.springframework.context.annotation.ConfigurationClassParser.processConfigurationClass(ConfigurationClassParser.java:245)
	at org.springframework.context.annotation.ConfigurationClassParser.processImports(ConfigurationClassParser.java:636)
	at org.springframework.context.annotation.ConfigurationClassParser.doProcessConfigurationClass(ConfigurationClassParser.java:303)
	at org.springframework.context.annotation.ConfigurationClassParser.processConfigurationClass(ConfigurationClassParser.java:245)
	at org.springframework.context.annotation.ConfigurationClassParser.processImports(ConfigurationClassParser.java:636)
	at org.springframework.context.annotation.ConfigurationClassParser.lambda$processDeferredImportSelectors$2(ConfigurationClassParser.java:566)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.springframework.context.annotation.ConfigurationClassParser.processDeferredImportSelectors(ConfigurationClassParser.java:563)
	at org.springframework.context.annotation.ConfigurationClassParser.parse(ConfigurationClassParser.java:188)
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.processConfigBeanDefinitions(ConfigurationClassPostProcessor.java:316)
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.postProcessBeanDefinitionRegistry(ConfigurationClassPostProcessor.java:233)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanDefinitionRegistryPostProcessors(PostProcessorRegistrationDelegate.java:271)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:91)
	at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:692)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:530)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:142)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:754)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:386)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:307)
	at com.mycompany.myapp.MyappApp.main(MyappApp.java:63)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.boot.devtools.restart.RestartLauncher.run(RestartLauncher.java:49)
Caused by: java.io.FileNotFoundException: class path resource [com/google/common/base/Supplier.class] cannot be opened because it does not exist
	at org.springframework.core.io.ClassPathResource.getInputStream(ClassPathResource.java:180)
	at org.springframework.core.type.classreading.SimpleMetadataReader.<init>(SimpleMetadataReader.java:51)
	at org.springframework.core.type.classreading.SimpleMetadataReaderFactory.getMetadataReader(SimpleMetadataReaderFactory.java:103)
	at org.springframework.boot.type.classreading.ConcurrentReferenceCachingMetadataReaderFactory.createMetadataReader(ConcurrentReferenceCachingMetadataReaderFactory.java:88)
	at org.springframework.boot.type.classreading.ConcurrentReferenceCachingMetadataReaderFactory.getMetadataReader(ConcurrentReferenceCachingMetadataReaderFactory.java:75)
	at org.springframework.core.type.classreading.SimpleMetadataReaderFactory.getMetadataReader(SimpleMetadataReaderFactory.java:81)
	at org.springframework.context.annotation.ConfigurationClassParser.asSourceClass(ConfigurationClassParser.java:732)
	at org.springframework.context.annotation.ConfigurationClassParser$SourceClass.getInterfaces(ConfigurationClassParser.java:961)
	at org.springframework.context.annotation.ConfigurationClassParser.processInterfaces(ConfigurationClassParser.java:376)
	at org.springframework.context.annotation.ConfigurationClassParser.processInterfaces(ConfigurationClassParser.java:384)
	at org.springframework.context.annotation.ConfigurationClassParser.doProcessConfigurationClass(ConfigurationClassParser.java:324)
	at org.springframework.context.annotation.ConfigurationClassParser.processConfigurationClass(ConfigurationClassParser.java:245)
	at org.springframework.context.annotation.ConfigurationClassParser.parse(ConfigurationClassParser.java:194)
	at org.springframework.context.annotation.ConfigurationClassParser.doProcessConfigurationClass(ConfigurationClassParser.java:296)
	at org.springframework.context.annotation.ConfigurationClassParser.processConfigurationClass(ConfigurationClassParser.java:245)
	at org.springframework.context.annotation.ConfigurationClassParser.processImports(ConfigurationClassParser.java:636)
	... 26 common frames omitted
```

**Motivation for or Use Case**

Its big bug for me. The reasons are described below. 

When you generate your new project. As a user, the first thing you expect is run the application and see it on http://localhost:8080 and explore the UI, APIs etc.

But here it fails to run the application and you see the big exception stackTrace instead. 
This is very annoying for any developer who is new to the JHipster/SpringBoot tech stack. The issue is also in the version 5.7.0

But this fails in this main purpose and it also against the JHipster vision:
- JHipster is awesome because it makes developer job easy and generates the Spring boot and Angular/React application out of the box when you answer initial questions while generating your project using "jhipster" command.

- JHipster generates production-ready project without any hassle in minutes. Its main selling point for the JHipster. But here it fails in achieving this gaol.

- Infact this is not good when someone want to try JHipster first time and that too with the latest JHipster version 5.7.1. 

P.S:
This happened to us when we were introducing JHipster to our new joinees.  

##### **Reproduce the error**

Step 1) Install latest JHipster v5.7.1 
Step 2) Create a new folder "newapp"
Step 3) Go inside the new folder named "newapp"
Step 4) Run the command "jhipster"
Step 5) Answer the questions as below:
? Which *type* of application would you like to create? Monolithic application (recommended for simple projects)
? What is the base name of your application? newapp
? What is your default Java package name? com.mycompany.newapp
? Do you want to use the JHipster Registry to configure, monitor and scale your application? No
? Which *type* of authentication would you like to use? JWT authentication (stateless, with a token)
? Which *type* of database would you like to use? SQL (H2, MySQL, MariaDB, PostgreSQL, Oracle, MSSQL)
? Which *production* database would you like to use? MySQL
? Which *development* database would you like to use? H2 with in-memory persistence
? Do you want to use the Spring cache abstraction? Yes, with the Ehcache implementation (local cache, for a single node)
? Do you want to use Hibernate 2nd level cache? No
? Would you like to use Maven or Gradle for building the backend? Maven
? Which other technologies would you like to use? 
? Which *Framework* would you like to use for the client? Angular
? Would you like to enable *SASS* stylesheet preprocessor? Yes
? Would you like to enable internationalization support? No
? Besides JUnit and Jest, which testing frameworks would you like to use? Cucumber
? Would you like to install other generators from the JHipster Marketplace? No

Step 6) Run the command "./mvnw"

Result: Application run fails with the exception stack Trace
org.springframework.beans.factory.BeanDefinitionStoreException: Failed to process import candidates for configuration class [

**Related issues**
Nothing found on issue list

##### **Suggest a Fix**

I have fixed the issue by adding below dependency in the pom.xml

```
<dependency>
    <groupId>com.google.guava</groupId>
    <artifactId>guava</artifactId>
    <version>27.0.1-jre</version>
</dependency>
```

##### **JHipster Version(s)**

Latest JHipster version 5.7.1. The problem also exists for the version 5.7.0
I have not tested this 5.6.0 and few previous versions. Last when I tested with JHipster 5 -it was working fine.

##### **JHipster configuration**

Please find the jhipster info command result at the start of the issue.

Here are my .yo-rc.json details
```
{
    "generator-jhipster": {
        "promptValues": {
            "packageName": "com.mycompany.newapp"
        },
        "jhipsterVersion": "5.7.1",
        "applicationType": "monolith",
        "baseName": "newapp",
        "packageName": "com.mycompany.newapp",
        "packageFolder": "com/mycompany/newapp",
        "serverPort": "8080",
        "authenticationType": "jwt",
        "cacheProvider": "ehcache",
        "enableHibernateCache": false,
        "websocket": false,
        "databaseType": "sql",
        "devDatabaseType": "h2Memory",
        "prodDatabaseType": "mysql",
        "searchEngine": false,
        "messageBroker": false,
        "serviceDiscoveryType": false,
        "buildTool": "maven",
        "enableSwaggerCodegen": false,
        "clientFramework": "angularX",
        "useSass": true,
        "clientPackageManager": "npm",
        "testFrameworks": ["cucumber"],
        "jhiPrefix": "jhi",
        "otherModules": [],
        "enableTranslation": false
    }
}
```

**Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**
No entities generated nor JDL generation used. These are second steps for me when the application runs fine.

**Browsers and Operating System**
Ubuntu 18.04 LTS 
No browser issue. Its backend issue in the SpringBoot
-   [ checked] Checking this box is mandatory (this is just to show you read everything)

Sure. We love JHipster. Our first choice is JHipster for our APIs and microservice architecture. Thank you!

**Comment by pmverma** [comment]
FYI, similar issue has been asked on gitter channel.
[https://gitter.im/jhipster/generator-jhipster?at=5b68680604436a1ae623976c](https://gitter.im/jhipster/generator-jhipster?at=5b68680604436a1ae623976c)

**Comment by ruddell** [comment]
I generated an app using your config, and running `./mvnw dependency:tree`, I see `com.google.guava:guava:jar:25.1-jre:compile` listed under Springfox's dependencies. I can also access Swagger without any issues (tested `dev` and `prod`).  Did you customize the project in any way?

**Comment by msshitole** [comment]
@ruddell Nopes. No customization did from my side. Created a new folder and generated a simple monolith project with the above Q&A. And run the application with the ./mvnw 

As you rightly said, the dependency is related to Springfox.

**Comment by atomfrede** [comment]
@ruddell @msshitole Same for me. Created a plain application with the same configuration as yours and it works just fine (I am pretty sure our CI pipeline would have detected that). Can you do a `mvn dependency:tree` and check which guava version is on your classpath and where it comes from?

**Comment by pmverma** [comment]
The only difference is the JDK version. I have not tested it though. Can you please downgrade the jdk version and try?

**Comment by msshitole** [comment]
@atomfrede guava version is (com.google.guava:guava:jar:25.1-jre) and it comes from the parent (io.springfox:springfox-swagger2:jar:2.9.2) 

Here is the snippet of **mvn dependency:tree**

> +- io.springfox:springfox-swagger2:jar:2.9.2:compile
[INFO] |  +- io.swagger:swagger-annotations:jar:1.5.20:compile
[INFO] |  +- io.swagger:swagger-models:jar:1.5.20:compile
[INFO] |  +- io.springfox:springfox-spi:jar:2.9.2:compile
[INFO] |  |  \- io.springfox:springfox-core:jar:2.9.2:compile
[INFO] |  +- io.springfox:springfox-schema:jar:2.9.2:compile
[INFO] |  +- io.springfox:springfox-swagger-common:jar:2.9.2:compile
[INFO] |  +- io.springfox:springfox-spring-web:jar:2.9.2:compile
[INFO] |  +- com.google.guava:guava:jar:25.1-jre:compile
[INFO] |  |  +- org.checkerframework:checker-qual:jar:2.0.0:compile
[INFO] |  |  +- com.google.errorprone:error_prone_annotations:jar:2.1.3:compile
[INFO] |  |  +- com.google.j2objc:j2objc-annotations:jar:1.1:compile
[INFO] |  |  \- org.codehaus.mojo:animal-sniffer-annotations:jar:1.14:compile

**Comment by msshitole** [comment]
@pmverma My version is Java 8 and do you want me to downgrade it further to which version?

> openjdk version "1.8.0_191"
OpenJDK Runtime Environment (build 1.8.0_191-8u191-b12-0ubuntu0.18.04.1-b12)
OpenJDK 64-Bit Server VM (build 25.191-b12, mixed mode)

**Comment by msshitole** [comment]
The issue has been fixed and it works fine now. The root cause of the problem was maven-compiler-plugin is failed to read guaa dependency. 

Here is the error log from the ./mvnw console.

> [ERROR] error reading /home/shitole/.m2/repository/com/google/guava/guava/25.1-jre/guava-25.1-jre.jar; zip file is empty

To fix this issue, I have manually went to my maven repository and deleted guava folder and regenerated the new project with the same steps and it works fine. Just for clarity, deleted below directory and regenerated project.

> /home/shitole/.m2/repository/com/google/guava

@ruddell and @atomfrede : You are right! Its not an universal issue. I have tested this with my team member as well and it works in his system without any issue. He has same Ubuntu OS and other configurations like me. 

Note: The resolution for this issue is independent to the JHipster framework. Hence we can close this issue. Though few people are facing this issue and root cause might be lying with the currupt guava jar.

Thank you all for looking into this and giving your inputs.
